File: frontend-app/src/types/cliente.ts
import type { Page } from './produto'
import type { TipoDocumento } from '../utils/documento'

export type { Page, TipoDocumento }

export interface Cliente {
  id: number
  nomeCompleto: string
  tipoDocumento: TipoDocumento
  numeroDocumento: string
  email: string
  codigoPais: string
  celular: string
  cep: string | null
  logradouro: string | null
  numero: string | null
  complemento: string | null
  bairro: string | null
  cidade: string | null
  estado: string | null
  ativo: boolean
  createdAt: string
  updatedAt: string
}

export interface ClienteRequest {
  nomeCompleto: string
  tipoDocumento: TipoDocumento
  numeroDocumento: string
  email: string
  /** Código ISO do país do celular (ex.: BR, US, PT). */
  codigoPais: string
  celular: string
  cep?: string | null
  logradouro?: string | null
  numero?: string | null
  complemento?: string | null
  bairro?: string | null
  cidade?: string | null
  estado?: string | null
}

export interface CepConsulta {
  cep: string
  logradouro: string
  bairro: string
  cidade: string
  estado: string
}

export type { PaisTelefoneOption } from '../utils/telefone'
export { PAISES_TELEFONE, PAIS_PADRAO } from '../utils/telefone'
export { TIPOS_DOCUMENTO, TIPO_DOCUMENTO_PADRAO } from '../utils/documento'

File: cerebro-backend/src/main/java/com/omnicore/cerebro_backend/model/Cliente.java
package com.omnicore.cerebro_backend.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tb_cliente", uniqueConstraints = {
        @UniqueConstraint(columnNames = { "tipo_documento", "numero_documento" })
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "nome_completo", nullable = false, length = 150)
    private String nomeCompleto;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_documento", nullable = false, length = 30)
    private TipoDocumento tipoDocumento = TipoDocumento.CPF;

    @Column(name = "numero_documento", nullable = false, length = 30)
    private String numeroDocumento;

    @Column(nullable = false, length = 150)
    private String email;

    @Builder.Default
    @Column(name = "codigo_pais", nullable = false, length = 4)
    private String codigoPais = "BR";

    @Column(nullable = false, length = 15)
    private String celular;

    @Column(length = 8)
    private String cep;

    @Column(length = 150)
    private String logradouro;

    @Column(length = 20)
    private String numero;

    @Column(length = 100)
    private String complemento;

    @Column(length = 100)
    private String bairro;

    @Column(length = 100)
    private String cidade;

    @Column(length = 2)
    private String estado;

    @Builder.Default
    @Column(nullable = false)
    private Boolean ativo = true;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
